tests/kokkos-based/helpers.hpp

```diff
@@ -84,5 +84,235 @@ auto create_stdvector_and_copy_rowwise(T sourceView)
   return result;
 }
 
+// create rank-1 mdspan (vector)
+template <typename ValueType,
+          typename mdspan_t = typename _blas2_signed_fixture<ValueType>::mdspan_r1_t>
+mdspan_t make_mdspan(ValueType *data, std::size_t ext) {
+  return mdspan_t(data, ext);
+}
+
+template <typename ValueType,
+          typename mdspan_t = typename _blas2_signed_fixture<const ValueType>::mdspan_r1_t>
+mdspan_t make_mdspan(const ValueType *data, std::size_t ext) {
+  return mdspan_t(data, ext);
+}
+
+template <typename ValueType>
+auto make_mdspan(std::vector<ValueType> &v) {
+  return make_mdspan(v.data(), v.size());
+}
+
+template <typename ValueType>
+auto make_mdspan(const std::vector<ValueType> &v) {
+  return make_mdspan(v.data(), v.size());
+}
+
+// create rank-2 mdspan (matrix)
+template <typename ValueType,
+          typename mdspan_t = typename _blas2_signed_fixture<ValueType>::mdspan_r2_t>
+mdspan_t make_mdspan(ValueType *data, std::size_t ext0, std::size_t ext1) {
+  return mdspan_t(data, ext0, ext1);
+}
+
+// no-tolerance (exact) comparison
+// TODO: add tolerance based comparison if needed (see is_same_matrix below)
+template <typename ElementType1,
+          typename LayoutPolicy1,
+          typename AccessorPolicy1,
+          typename ElementType2,
+          typename LayoutPolicy2,
+          typename AccessorPolicy2>
+bool is_same_vector(
+    const mdspan<ElementType1, extents<dynamic_extent>, LayoutPolicy1, AccessorPolicy1> &v1,
+    const mdspan<ElementType2, extents<dynamic_extent>, LayoutPolicy2, AccessorPolicy2> &v2)
+{
+  const auto size = v1.extent(0);
+  if (size != v2.extent(0))
+    return false;
+  const auto v1_view = KokkosKernelsSTD::Impl::mdspan_to_view(v1);
+  const auto v2_view = KokkosKernelsSTD::Impl::mdspan_to_view(v2);
+  int diff = false;
+  Kokkos::parallel_reduce(size,
+    KOKKOS_LAMBDA(const std::size_t i, decltype(diff) &d){
+        d = d || !(v1_view(i) == v2_view(i));
```

Comment:
Would you consider using the `LOr` reducer to document intent, instead of rolling the reduction by hand?  (The element access function would then just be `v1_view(i) != v2_view(i)`.)

https://github.com/kokkos/kokkos/blob/2834f94af9b01debf67c1aaa3f0eb0c903d72c8d/core/src/Kokkos_Parallel_Reduce.hpp#L301

Also, would you consider using `operator[]` instead of `operator()` to access the view?  That works now, and would be more consistent with C++23.